### PR TITLE
Implement flags grouping for karmada-scheduler-estimator

### PR DESCRIPTION
**Pull request by AllenZMC**
**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:
Implement flags grouping for karmada-scheduler-estimator #1485
**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Comment by lonelyCZ** [comment]
There seems some problems, let's to check it.

```
[root@master67 karmada]# ./karmada-scheduler-estimator -h
generic flag redefined: kubeconfig
panic: generic flag redefined: kubeconfig

goroutine 1 [running]:
github.com/spf13/pflag.(*FlagSet).AddFlag(0xc0003bac00, 0xc0000af900)
	/root/lonelyCZ/karmada-lonelyCZ/karmada/vendor/github.com/spf13/pflag/flag.go:848 +0x5f6
github.com/spf13/pflag.(*FlagSet).VarPF(0x1, {0x1f17750, 0xc000557380}, {0x1c939a4, 0xa}, {0x0, 0x0}, {0x1ce1812, 0x36})
	/root/lonelyCZ/karmada-lonelyCZ/karmada/vendor/github.com/spf13/pflag/flag.go:831 +0x151
github.com/spf13/pflag.(*FlagSet).VarP(...)
	/root/lonelyCZ/karmada-lonelyCZ/karmada/vendor/github.com/spf13/pflag/flag.go:837
github.com/spf13/pflag.(*FlagSet).StringVar(0x0, 0x1c8f814, {0x1c939a4, 0x1766466}, {0x0, 0x40d407}, {0x1ce1812, 0xa72c60})
	/root/lonelyCZ/karmada-lonelyCZ/karmada/vendor/github.com/spf13/pflag/string.go:37 +0x69
github.com/karmada-io/karmada/cmd/scheduler-estimator/app/options.(*Options).AddFlags(0xc000557380, 0xc000118360)
	/root/lonelyCZ/karmada-lonelyCZ/karmada/cmd/scheduler-estimator/app/options/options.go:40 +0x65
github.com/karmada-io/karmada/cmd/scheduler-estimator/app.NewSchedulerEstimatorCommand({0x1f2b570, 0xc00043ea00})
	/root/lonelyCZ/karmada-lonelyCZ/karmada/cmd/scheduler-estimator/app/scheduler-estimator.go:53 +0x15d
main.runSchedulerEstimatorCmd()
	/root/lonelyCZ/karmada-lonelyCZ/karmada/cmd/scheduler-estimator/main.go:23 +0x4f
main.main()
	/root/lonelyCZ/karmada-lonelyCZ/karmada/cmd/scheduler-estimator/main.go:13 +0x19
```

**Comment by AllenZMC** [comment]
I will check it later.

**Comment by AllenZMC** [comment]
@lonelyCZ , after I modified a `kubeconfig` flag name to the `kubeconfig1` flag name
```
      --kubeconfig string                                                                                                        
                Paths to a kubeconfig. Only required if out-of-cluster.
      --kubeconfig1 string                                                                                                       
                Path to a KubeConfig. Only required if out-of-cluster.

```

**Comment by AllenZMC** [comment]
And I commented out a line of code and there was no error 
	```
genericFlagSet := fss.FlagSet("generic")
//genericFlagSet.AddGoFlagSet(flag.CommandLine)
opts.AddFlags(genericFlagSet)
```

File: cmd/scheduler-estimator/app/scheduler-estimator.go [hunk 1]
```diff
@@ -42,9 +46,22 @@ func NewSchedulerEstimatorCommand(ctx context.Context) *cobra.Command {
 		},
 	}
 
-	opts.AddFlags(cmd.Flags())
+	fss := cliflag.NamedFlagSets{}
+
+	genericFlagSet := fss.FlagSet("generic")
+	genericFlagSet.AddGoFlagSet(flag.CommandLine)
```
Comment (lonelyCZ) [inline cmd/scheduler-estimator/app/scheduler-estimator.go:52]: > I find it in vendor sigs.k8s.io/controller-runtime/pkg/client/config/config.go
> ```
> func init() {
>	// TODO: Fix this to allow double vendoring this library but still register flags on behalf of users
>	flag.StringVar(&kubeconfig, "kubeconfig", "",
>		"Paths to a kubeconfig. Only required if out-of-cluster.")
> }
> ```

Do we need to this default value? If not, we don't need to set `genericFlagSet.AddGoFlagSet(flag.CommandLine)`. @Garrybest

Comment (Garrybest) [reply cmd/scheduler-estimator/app/scheduler-estimator.go:52]: The `scheduler-estimator` does not use `controller-runtime` to build. So this flag has nothing to do with it.

Comment (RainbowMango) [reply cmd/scheduler-estimator/app/scheduler-estimator.go:52]: My question is why the `--kubeconfig` flag will be added by controller-runtime, seems `scheduler estimater` didn't import controller-runtime?

Comment (AllenZMC) [reply cmd/scheduler-estimator/app/scheduler-estimator.go:52]:
```
func run(ctx context.Context, opts *options.Options) error {
	klog.Infof("karmada-scheduler-estimator version: %s", version.Get())
	go serveHealthzAndMetrics(net.JoinHostPort(opts.BindAddress, strconv.Itoa(opts.SecurePort)))

	restConfig, err := clientcmd.BuildConfigFromFlags(opts.Master, opts.KubeConfig)
       ...
```

Comment (lonelyCZ) [reply cmd/scheduler-estimator/app/scheduler-estimator.go]: I guess it is imported from other package that `scheduler-estimator` invokes.

Comment (RainbowMango) [reply cmd/scheduler-estimator/app/scheduler-estimator.go]: That's possible. But, where?

Comment (AllenZMC) [reply cmd/scheduler-estimator/app/scheduler-estimator.go]: I am also very curious.

Comment (AllenZMC) [reply cmd/scheduler-estimator/app/scheduler-estimator.go]: I have modified it correctly

Comment (lonelyCZ) [reply cmd/scheduler-estimator/app/scheduler-estimator.go]: Ok, thank you. @AllenZMC

**Comment by lonelyCZ** [comment]
Seemingly, there is not CI test in this component? @RainbowMango @Garrybest

**Comment by Garrybest** [comment]
Right. The estimator and descheduler do not have e2e tests, only unit tests. 🤣

**Comment by RainbowMango** [comment]
LGTM by the way

**Comment by AllenZMC** [comment]
> /lgtm

approve please😀

**Review by RainbowMango — APPROVED** [review]
/approve
on behalf of @Garrybest

**Pull request merged**